stan/math/rev/fun/multiply.hpp

```diff
@@ -29,41 +29,43 @@ template <typename T1, typename T2, require_all_matrix_t<T1, T2>* = nullptr,
 inline auto multiply(const T1& A, const T2& B) {
   check_multiplicable("multiply", "A", A, "B", B);
   if (!is_constant<T2>::value && !is_constant<T1>::value) {
-    arena_t<promote_scalar_t<var, T1>> arena_A = to_ref(A);
-    arena_t<promote_scalar_t<var, T2>> arena_B = to_ref(B);
-    arena_t<promote_scalar_t<double, T1>> arena_A_val = value_of(arena_A);
-    arena_t<promote_scalar_t<double, T2>> arena_B_val = value_of(arena_B);
+    arena_t<promote_scalar_t<var, T1>> arena_A = A;
+    arena_t<promote_scalar_t<var, T2>> arena_B = B;
     using return_t
-        = return_var_matrix_t<decltype(arena_A_val * arena_B_val), T1, T2>;
-    arena_t<return_t> res = arena_A_val * arena_B_val;
+        = return_var_matrix_t<decltype(arena_A.val_op() * arena_B.val_op()), T1,
+                              T2>;
+    arena_t<return_t> res = arena_A.val_op() * arena_B.val_op();
```

Comment:
For `var<matrix>` this will def be faster. tbh I'm not sure if pulling out those is faster for `mat<var>` multiplication. It forces some big allocations so while we will get better access and will be able to use SIMD instructions I think at large'ish NxM those copes are going to be very noticeable

Reply:
Large NxM is where the copies aren't going to matter (cause the computation will be the expensive thing).

For `var<matrix>` I didn't think there would be a copy. The `arena_t` created will just reference the one inside the `var_value<matrix>`: https://github.com/stan-dev/math/blob/develop/stan/math/rev/core/arena_matrix.hpp#L80

Reply:
Oh ya know I think I can just use `to_arena()` here, then for matrix var it will make the copy and var matrix it will be just a copy of the pointer